### PR TITLE
Feature/rolling filter

### PR DESCRIPTION
**Pull request by dtim1985**
Add rolling filter
There is no pooling for rolling filter - it is easy to implement it on bidder side so it can use both client pool and rolling filter pool
Some refactoring in tests and KeySet
The KeySet is probably will be changed according to a particular usage

### CODE REVIEW

File: utils/testutils/bloomd_addrs.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	"net/url"
+	"os"
+	"strings"
+	"sync"
+	"testing"
+)
+
+var BloomdAddrs = addrsFromEnv
+
+func UseAddrsFromEnv() {
```
Comment (antnsi) [inline utils/testutils/bloomd_addrs.go]: is it used?

Comment (dtim1985) [reply utils/testutils/bloomd_addrs.go]: No. I was thinking to move a bloomd mock as well. I'll delete it for now

File: extensions/rolling/filter.go [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+	unit   string
+	client *bloomd.Client
+	rs     resultsSet
+}
+
+// NewFilter creates a new Filter
+// unit - unit of time, e.g. Week, Day or Month. All keys being set during period with a same unit will be stored in a single filter.
+// period - period in specified time units to consider in check operations
+// namer - provides algorithm to name filters according to specified unit of time
+func NewFilter(namer Namer, unit string, period int, client *bloomd.Client) *Filter {
```
Comment (orktes) [inline extensions/rolling/filter.go]: custom type for `unit` might be nice

**Comment by orktes** [comment]
Looks good! Btw adding `(t/b).Helper()` into the beginning of test helpers produces nicer stack traces! :)

**Pull request merged**